### DISCUSSION TITLE
Gradle and ignoring a repository via hostRules

### DISCUSSION THREAD

**Discussion by jamesdh** [renovatebot/renovate#12901] [Request Help] [answered]
Our renovate self-hosted runner is taking a considerable amount of time to check our gradle tasks because it appears to be repeatedly trying to resolve against a repository that has very few of the dependencies, and a private repository that requires authentication. 

I've tried disabling the two repositories via hostRules, a la: 

```
"hostRules": [
  {
    "matchHost": "https://maven.pkg.github.com",
    "enabled": false
  },
  {
    "matchHost": "http://maven.openimaj.org",
    "enabled": false
  }
}
```

I also tried using just the hostname form for both of them. Neither is preventing lookups against them. 

Am I missing something obvious, or is this a feature that's just not functional w/ the gradle datasource at the moment?

**Comment by rarkins** [comment, marked as answer]
Have you tried overriding the detected `registryUrls` using a `packageRules` rule?

**Reply by jamesdh** [reply]
Just tried adding the following to our gradle packageRules: 

```yaml
"registryUrls": [
  "https://repo.maven.apache.org/maven2/",
  "https://plugins.gradle.org/m2"
]
```

But I'm still seeing an excessive amount of resolutions against other registries, a la: 
> DEBUG: Content is not found for Maven url (repository=...)
       "url": "https://maven.pkg.github.com/proximal-health/platform/ch/qos/logback/logback-classic/maven-metadata.xml",
       "statusCode": undefined

**Reply by rarkins** [reply]
Please share the whole config, or better yet a public reproduction repo

**Reply by jamesdh** [reply]
Ok, I was trying to match on the gradle manager (e.g. `"matchManagers": ["gradle"]`) but when I switched it to `"matchDatasources": ["maven"]` it appears to be working!

I might add though that when you only touch the renovate configs every few months, it can be very confusing to understand the differences between datasources, managers, etc, especially when some of them share the same name. Might be really helpful (and save you some time @rarkins) if the docs could clarify that a bit better.
